packages/data/src/world/mq/deku_tree_mq.yml

```diff
@@ -2,126 +2,282 @@
     "MQ Deku Tree Grass Entrance Lower 3": "can_cut_grass"
     "MQ Deku Tree Grass Entrance Lower 4": "can_cut_grass"
     "MQ Deku Tree Grass Entrance Lower 5": "can_cut_grass"
+"MQ Deku Tree 2nd Floor":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree Lobby": "true"
+    "MQ Deku Tree 3rd Floor": "true"
+    "MQ Deku Tree Room Before Compass Room Entry Door": "has_fire || (can_use_sticks && event(DEKU_MQ_MAIN_TORCH)) || (event(DEKU_MQ_MAIN_TORCH) && can_use_bow)"
+  locations:
+    "MQ Deku Tree Map Chest": "true"
+    "MQ Deku Tree GS Lobby Crate": "gs && can_damage_skull"
+    "MQ Deku Tree Heart Lobby": "true"
     "MQ Deku Tree Grass Entrance Upper 1": "can_cut_grass"
     "MQ Deku Tree Grass Entrance Upper 2": "can_cut_grass"
-    "MQ Deku Tree Grass Slingshot Room Front 1": "can_cut_grass"
-    "MQ Deku Tree Grass Slingshot Room Front 2": "can_cut_grass"
-    "MQ Deku Tree Grass Slingshot Room Back 1": "can_cut_grass"
-    "MQ Deku Tree Grass Slingshot Room Back 2": "can_cut_grass"
-    "MQ Deku Tree Grass Room Before Compass 1": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
-    "MQ Deku Tree Grass Room Before Compass 2": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
-    "MQ Deku Tree Grass Room Before Compass 3": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
-    "MQ Deku Tree Grass Room Before Compass 4": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
-    "MQ Deku Tree Grass Room Before Compass 5": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
-    "MQ Deku Tree Grass Room Before Compass 6": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
-    "MQ Deku Tree Grass Room Before Compass 7": "can_cut_grass && (can_use_bow || can_use_sticks || can_use_din)"
-    "MQ Deku Tree Grass Basement Lower 1": "can_cut_grass"
-    "MQ Deku Tree Grass Basement Lower 2": "can_cut_grass"
-    "MQ Deku Tree Grass Basement Lower 3": "can_cut_grass"
-    "MQ Deku Tree Grass Basement Lower 4": "can_cut_grass"
-    "MQ Deku Tree Grass Room Before Spike 1": "can_cut_grass && (can_use_bow || (can_use_slingshot && (can_use_sticks || has_fire)))"
-    "MQ Deku Tree Grass Room Before Spike 2": "can_cut_grass && (can_use_bow || (can_use_slingshot && (can_use_sticks || has_fire)))"
-    "MQ Deku Tree Grass Room Before Spike 3": "can_cut_grass && (can_use_bow || (can_use_slingshot && (can_use_sticks || has_fire)))"
-    "MQ Deku Tree Grass Room Before Spike 4": "can_cut_grass && (can_use_bow || (can_use_slingshot && (can_use_sticks || has_fire)))"
-    "MQ Deku Tree Heart Lobby": "true"
-    "MQ Deku Tree Heart Before Compass": "can_use_bow || can_use_sticks || can_use_din"
-    "MQ Deku Tree Heart Slingshot Room": "true"
-"Deku Tree Compass Room":
+"MQ Deku Tree Room Before Compass Room Entry Door":
+  dungeon: DT
+  exits:
+    "MQ Deku Tree 2nd Floor": "true"
+    "MQ Deku Tree Room Before Compass Room Door to 2nd Floor": "true"
```

Comment:
Do we know yet if this is actually the case? Ghun has been testing, so he should have been able to come through this door right? Are you behind the web?

Reply:
Yes, you appear behind the web. The only place I am not sure about where you spawn from a door is Jabu-Jabu MQ Basement Side Room > Basement (the red tentacle looks to be blocking going to side room, but I don't know if it blocks going to basement main).